### PR TITLE
Accept tries counter for query functions

### PR DESCRIPTION
**Pull request by kevinkjt2000**
Per [Discord from @ItsDrike](https://discord.com/channels/936788458939224094/938623566964981802/1150384415592153118), 
> hm, I just noticed that we don't take **kwargs in JavaServer.query, and pass them over to _retry_query, like we do with status, this makes it impossible to do JavaServer.query(tries=1), whcih would be caught by the retry decorator, forcing people to always make 3 retires.

This should fix that.

### CODE REVIEW

File: mcstatus/server.py
Comment (kevinkjt2000) [inline mcstatus/server.py:163]: I realize it looks silly to have an unused argument, but `Callable.kwargs` is unmodifiable for now. This PEP mentions that support may be added in the future https://peps.python.org/pep-0612/#concatenating-keyword-parameters
And until that's the case, there is not a way for typing tools like pyright to know that our retry decorator pops a keyword argument out of the `Callable`.

**Review by ItsDrike — APPROVED** [review]
It might actually be a good idea to make `tries` an explicit kwarg for status too, rather than just using `**kwargs` there blindly. The reason this isn't done is because the kwargs are being passed over to the `ServerPinger` class's `__init__`, but we can still take `tries` explicitly, and also take extra kwargs for this. The reason this would be better is that editors would let you know that there is a `tries` argument, as opposed to just letting you know that you can pass some `**kwargs`.

There is one important note about doing this though, which is that by giving `tries` a default value of 3, this will always override the `tries=3` argument passed to the actual decorator, whereas if the kwarg wasn't passed at all, that would be the default instead. It doesn't really matter since it's just the same value (of 3) though. What we could do is have it default to `None` instead, and handle that in the decorator, or only use these kwargs, and don't make the decorator take `tries` directly.

Not sure these suggestions should block this PR though, since as far as this PR's goal goes, this does fulfill it. So I'm giving it an approval here, but it could be worth looking into these in another PR, or if you wish, feel free to add them here in some manner.

File: mcstatus/server.py
Comment (ItsDrike) [reply mcstatus/server.py:163]: I think it could be done with Concatenate if we make the `tries` argument required when using the decorator (see my approval comment for why we might actually want to do this). But I'm not sure pyright detects this, as the unused argument logic might be separated from that, so it's possible it would trigger unused warnings anyway.

Another way to solve it would be to make the decorator check for `_tries` instead of `tries`, and only make the user-facing (non-retry) functions take `tries`, but pass them over as `_tries=tires` when calling the retry func.

File: mcstatus/server.py
Comment (ItsDrike) [inline mcstatus/server.py:153]: Actually, there is one nitpick I'd make here, should `tries`  be keyword only argument here too?
Doing `JavaServer.query(1)` looks a bit weird to me as it's pretty unclear what that 1 is for.

```suggestion
    def query(self, *, tries: int = 3) -> QueryResponse:
```

Same remark for `async_query`

File: mcstatus/server.py
Comment (kevinkjt2000) [reply mcstatus/server.py:163]: That PEP talks about Concatenate making sense for modifying keyword arguments but follows up with why it will be tricky to implement. `Callable.kwargs` is not editable yet for the reasons outlined in the PEP.

Comment (ItsDrike) [reply mcstatus/server.py:163]: Oh right, kwargs aren't supported by `Concatenate`. I assumed the pep jut pointed to param specs as a whole instead of actually reading the linked section. Hm, that's annoying. In this case only the `_tries` suggestion makes sense here.

**Review by PerchunPak — APPROVED** [review]
lgtm

**Review by ItsDrike — APPROVED** [review]

**Review by PerchunPak — APPROVED** [review]

**Comment by PerchunPak** [comment]
So this PR is blocked only by one conversation, which is actually more a note. @kevinkjt2000 can we merge it?

**Pull request merged**
